### PR TITLE
Specific FFmpeg version is supported.

### PR DESCRIPTION
**Pull request by nineKnight**

### CODE REVIEW

File: media-autobuild_suite.bat [hunk 1]
```diff
@@ -688,6 +688,26 @@ if %buildffmpegUp%==3 set "ffmpegUpdate=onlyFFmpeg"
...
 if %deleteINI%==1 echo.ffmpegUpdate=^%buildffmpegUp%>>%ini%
 
+:ffmpegVer
+if %ffmpegVersionINI%==0 (
+    echo -------------------------------------------------------------------------------
+    echo -------------------------------------------------------------------------------
+    echo.
+    echo. Always build latest FFmpeg?
+    echo. 1 = Yes
+    echo. x.y.z = Specific FFmpeg version (the git tag, e.g. 4.2.2 for n4.2.2^)
```
Comment (wiiaboo) [inline media-autobuild_suite.bat]: No, this seems like a terrible ideia.
There's no verification at all of what the user writes.

File: build/media-suite_config.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  case $1 in
+    --ffmpegVersion=* ) ffmpegVersion="${1#*=}"; shift ;;
+    -- ) shift; break ;;
+    -* ) echo "Error, unknown option: '$1'."; exit 1 ;;
+    * ) break ;;
+  esac
+done
+
+[ "$ffmpegVersion" == '' ] && ffmpegVersion='latest'
+sed -ri '/readme.txt/{s/(ffmpeg-).*(-win)/\1'$ffmpegVersion'\2/g}' "$LOCALBUILDDIR"/media-suite_helper.sh
```
Comment (wiiaboo) [inline build/media-suite_config.sh]: No.
Why would you need to change suite files for this? This would break git-synced suite installations.

File: build/media-suite_compile.sh [hunk 3]
```diff
@@ -1916,7 +1927,7 @@ if [[ $ffmpeg != "no" ]]; then
             "$LOCALDESTDIR"/bin-video/{sw{scale,resample},postproc}.lib
             )
         _check=()
-        sedflags="prefix|bindir|extra-version|pkg-config-flags"
+        sedflags="prefix|bindir|extra-\w*|pkg-config-flags|disable-debug|shlibdir"
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: Why are you removing disable-debug? This is imperative to leave in the compiled binary for proper issues to reported to FFmpeg.
These modifications are unrelated to this PR and should proposed in a separate PR or issue.

File: build/media-suite_compile.sh [hunk 4]
```diff
@@ -1438,6 +1439,11 @@ if [[ $x264 != no ]]; then
         if [[ $standalone = y && $x264 =~ (full|fullv) ]]; then
             _check=("$LOCALDESTDIR"/opt/lightffmpeg/lib/pkgconfig/libav{codec,format}.pc)
             do_vcs "https://git.ffmpeg.org/ffmpeg.git"
+            if [[ $ffmpegVersion == latest ]]; then
+                echo > VERSION
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: No, this is also a terrible idea. Why are you forcing FFmpeg to use what's inside this file, which might not even match in anyway the current checked out branch/tag?

**Comment by wiiaboo** [comment]
A proposed solution to this was already thought about in https://github.com/m-ab-s/media-autobuild_suite/issues/1512#issuecomment-569697115.

Overall, this PR seems like a terrible implementation of that idea.

File: media-autobuild_suite.bat (hunk 1 above)
Comment (nineKnight) [reply media-autobuild_suite.bat]: No guarantee is needed. `do_vcs` will fail and log will be written.

File: build/media-suite_config.sh (hunk 2 above)
Comment (nineKnight) [reply build/media-suite_config.sh]: Is there a script for the suite to update?

File: build/media-suite_compile.sh (hunk 3 above)
Comment (nineKnight) [reply build/media-suite_compile.sh]: You can leave it as you wish.

File: build/media-suite_compile.sh (hunk 4 above)
Comment (nineKnight) [reply build/media-suite_compile.sh]: Do you really verify how it works?

**Comment by nineKnight** [comment]
@wiiaboo Your version is desired. Appreciate your work.

File: media-autobuild_suite.bat (hunk 1 above)
Comment (wiiaboo) [reply media-autobuild_suite.bat]: Yeah, the user does stupid things and we have to deal with it. That's not how it works.

File: build/media-suite_config.sh (hunk 2 above)
Comment (wiiaboo) [reply build/media-suite_config.sh]: Yes, and there's the fact that it's git.

File: build/media-suite_compile.sh (hunk 4 above)
Comment (wiiaboo) [reply build/media-suite_compile.sh]: Not 100%, but VERSION is used for release branches by FFmpeg, so anyway it would be wrong to write to it.

**Comment by wiiaboo** [comment]
This is a low-priority feature that will be implemented when someone else has time and interest for it. That's what you pay for with using open source.

File: build/media-suite_compile.sh (hunk 4 above)
Comment (nineKnight) [reply build/media-suite_compile.sh]: If `echo > VERSION`, FFmpeg won't use the `VERSION` file. So if not do it like this, how do you specify the version shown in `ffmpeg -version`, or do you always build the latest FFmpeg?

Comment (wiiaboo) [reply build/media-suite_compile.sh]: --extra-version, like I said somewhere else.

**Comment by nineKnight** [comment]
Yes, you are right.

在 2020年3月27日星期五，Ricardo Constantino <notifications@github.com> 写道：

**Comment by Barough** [comment]
Think it would be great if someone could make it possible to compile a specific ffmpeg version through the suite

File: build/media-suite_compile.sh (hunk 4 above)
Comment (nineKnight) [reply build/media-suite_compile.sh]: Yes, you are right.
